From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bacteriawa <A3167717663@hotmail.com>
Date: Wed, 22 Apr 2026 23:55:19 +0800
Subject: [PATCH] Leaf Remove streams in MobSensor

Co-authored by: Taiyou06 <kaandindar21@gmail.com>
As part of: Leaf (https://github.com/Winds-Studio/Leaf/blob/ver/1.21.11/leaf-server/minecraft-patches/features/0234-Remove-streams-in-MobSensor.patch)
Licensed under: MIT (https://github.com/Winds-Studio/Leaf/blob/ver/1.21.11/licenses/MIT.txt)

diff --git a/net/minecraft/world/entity/ai/sensing/MobSensor.java b/net/minecraft/world/entity/ai/sensing/MobSensor.java
index ac9c50d680814b5752f7c712633555d587633598..ccfe630c9bf9c5bc6948772ba703dc0282a247cb 100644
--- a/net/minecraft/world/entity/ai/sensing/MobSensor.java
+++ b/net/minecraft/world/entity/ai/sensing/MobSensor.java
@@ -46,10 +46,15 @@ public class MobSensor<T extends LivingEntity> extends Sensor<T> {
     public void checkForMobsNearby(final T body) {
         Optional<List<LivingEntity>> livingEntitiesMemory = body.getBrain().getMemory(MemoryModuleType.NEAREST_LIVING_ENTITIES);
         if (!livingEntitiesMemory.isEmpty()) {
-            boolean mobPresent = livingEntitiesMemory.get().stream().anyMatch(entity -> this.mobTest.test(body, entity));
-            if (mobPresent) {
-                this.mobDetected(body);
+            // Leaf start - Remove streams in MobSensor
+            List<LivingEntity> entities = livingEntitiesMemory.get();
+            for (int i = 0, size = entities.size(); i < size; i++) {
+                if (this.mobTest.test(body, entities.get(i))) {
+                    this.mobDetected(body);
+                    break;
+                }
             }
+            // Leaf end - Remove streams in MobSensor
         }
     }
 
